Unittest is not buildable with c++17

### What version of gRPC and what language are you using?

* gRPC version: v1.55.0
* language: not relatived

### What operating system (Linux, Windows,...) and version?

Ubuntu 20.04

### What runtime / compiler are you using (e.g. python version or version of gcc)

gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) 

### What did you do?

I try to build all unittest with flag `-DCMAKE_CXX_STANDARD=17`

```bash
cmake -GNinja \
    -DCMAKE_CXX_STANDARD=17 \
    -DCMAKE_BUILD_TYPE=Release \
    -DgRPC_BUILD_TESTS=ON \
    ../grpc
ninja
```

### What did you expect to see?

Build ok at least.

### What did you see instead?

Compiler error:

```
FAILED: CMakeFiles/xds_client_test.dir/test/core/xds/xds_client_test.cc.o 
/usr/bin/c++ -DCARES_STATICLIB -I/root/projects/grpc/grpc/third_party/re2 -I/root/projects/grpc/grpc/third_party/zlib -I/root/projects/grpc/grpc -I/root/projects/grpc/grpc/include -I/root/projects/grpc/grpc/third_party/address_sorting/include -I/root/projects/grpc/build_test/third_party/re2 -I/root/projects/grpc/grpc/third_party/boringssl-with-bazel/src/include -I/root/projects/grpc/grpc/src/core/ext/upb-generated -I/root/projects/grpc/grpc/src/core/ext/upbdefs-generated -I/root/projects/grpc/grpc/third_party/upb -I/root/projects/grpc/grpc/third_party/utf8_range -I/root/projects/grpc/grpc/third_party/xxhash -I/root/projects/grpc/build_test/third_party/zlib -I/root/projects/grpc/grpc/third_party/googletest/googletest/include -I/root/projects/grpc/grpc/third_party/googletest/googletest -I/root/projects/grpc/grpc/third_party/googletest/googlemock/include -I/root/projects/grpc/grpc/third_party/googletest/googlemock -I/root/projects/grpc/build_test/gens -I/root/projects/grpc/grpc/third_party/protobuf/src -I/root/projects/grpc/grpc/third_party/abseil-cpp -I/root/projects/grpc/build_test/third_party/cares/cares -I/root/projects/grpc/grpc/third_party/cares/cares -I/root/projects/grpc/grpc/third_party/cares/cares/include -O3 -DNDEBUG -std=gnu++17 -fPIE -pthread -MD -MT CMakeFiles/xds_client_test.dir/test/core/xds/xds_client_test.cc.o -MF CMakeFiles/xds_client_test.dir/test/core/xds/xds_client_test.cc.o.d -o CMakeFiles/xds_client_test.dir/test/core/xds/xds_client_test.cc.o -c /root/projects/grpc/grpc/test/core/xds/xds_client_test.cc
/root/projects/grpc/grpc/test/core/xds/xds_client_test.cc: In constructor ‘grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::Builder::Builder()’:
/root/projects/grpc/grpc/test/core/xds/xds_client_test.cc:161:33: error: no matching function for call to ‘std::optional<grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode>::emplace()’
  161 |       Builder() { node_.emplace(); }
      |                                 ^
In file included from /usr/include/c++/9/bits/node_handle.h:39,
                 from /usr/include/c++/9/bits/stl_tree.h:72,
                 from /usr/include/c++/9/map:60,
                 from /root/projects/grpc/grpc/src/core/ext/xds/xds_client.h:22,
                 from /root/projects/grpc/grpc/test/core/xds/xds_client_test.cc:21:
/usr/include/c++/9/optional:845:2: note: candidate: ‘template<class ... _Args> std::enable_if_t<is_constructible_v<_Tp, _Args&& ...>, _Tp&> std::optional<_Tp>::emplace(_Args&& ...) [with _Args = {_Args ...}; _Tp = grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode]’
  845 |  emplace(_Args&&... __args)
      |  ^~~~~~~
/usr/include/c++/9/optional:845:2: note:   template argument deduction/substitution failed:
In file included from /usr/include/c++/9/bits/move.h:55,
                 from /usr/include/c++/9/bits/stl_pair.h:59,
                 from /usr/include/c++/9/bits/stl_algobase.h:64,
                 from /usr/include/c++/9/bits/stl_tree.h:63,
                 from /usr/include/c++/9/map:60,
                 from /root/projects/grpc/grpc/src/core/ext/xds/xds_client.h:22,
                 from /root/projects/grpc/grpc/test/core/xds/xds_client_test.cc:21:
/usr/include/c++/9/type_traits: In substitution of ‘template<bool _Cond, class _Tp> using enable_if_t = typename std::enable_if::type [with bool _Cond = std::is_constructible_v<grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode>; _Tp = grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode&]’:
/usr/include/c++/9/optional:845:2:   required by substitution of ‘template<class ... _Args> std::enable_if_t<is_constructible_v<grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode, _Args&& ...>, grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode&> std::optional<grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode>::emplace<_Args ...>(_Args&& ...) [with _Args = {}]’
/root/projects/grpc/grpc/test/core/xds/xds_client_test.cc:161:33:   required from here
/usr/include/c++/9/type_traits:2378:11: error: no type named ‘type’ in ‘struct std::enable_if<false, grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode&>’
 2378 |     using enable_if_t = typename enable_if<_Cond, _Tp>::type;
      |           ^~~~~~~~~~~
In file included from /usr/include/c++/9/bits/node_handle.h:39,
                 from /usr/include/c++/9/bits/stl_tree.h:72,
                 from /usr/include/c++/9/map:60,
                 from /root/projects/grpc/grpc/src/core/ext/xds/xds_client.h:22,
                 from /root/projects/grpc/grpc/test/core/xds/xds_client_test.cc:21:
/usr/include/c++/9/optional:855:2: note: candidate: ‘template<class _Up, class ... _Args> std::enable_if_t<is_constructible_v<_Tp, std::initializer_list<_Up>&, _Args&& ...>, _Tp&> std::optional<_Tp>::emplace(std::initializer_list<_Up>, _Args&& ...) [with _Up = _Up; _Args = {_Args ...}; _Tp = grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode]’
  855 |  emplace(initializer_list<_Up> __il, _Args&&... __args)
      |  ^~~~~~~
/usr/include/c++/9/optional:855:2: note:   template argument deduction/substitution failed:
/root/projects/grpc/grpc/test/core/xds/xds_client_test.cc:161:33: note:   candidate expects at least 1 argument, 0 provided
  161 |       Builder() { node_.emplace(); }
      |                                 ^
/root/projects/grpc/grpc/test/core/xds/xds_client_test.cc: In member function ‘grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::Builder& grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::Builder::set_node_id(std::string)’:
/root/projects/grpc/grpc/test/core/xds/xds_client_test.cc:164:47: error: no matching function for call to ‘std::optional<grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode>::emplace()’
  164 |         if (!node_.has_value()) node_.emplace();
      |                                               ^
In file included from /usr/include/c++/9/bits/node_handle.h:39,
                 from /usr/include/c++/9/bits/stl_tree.h:72,
                 from /usr/include/c++/9/map:60,
                 from /root/projects/grpc/grpc/src/core/ext/xds/xds_client.h:22,
                 from /root/projects/grpc/grpc/test/core/xds/xds_client_test.cc:21:
/usr/include/c++/9/optional:845:2: note: candidate: ‘template<class ... _Args> std::enable_if_t<is_constructible_v<_Tp, _Args&& ...>, _Tp&> std::optional<_Tp>::emplace(_Args&& ...) [with _Args = {_Args ...}; _Tp = grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode]’
  845 |  emplace(_Args&&... __args)
      |  ^~~~~~~
/usr/include/c++/9/optional:845:2: note:   template argument deduction/substitution failed:
/usr/include/c++/9/optional:855:2: note: candidate: ‘template<class _Up, class ... _Args> std::enable_if_t<is_constructible_v<_Tp, std::initializer_list<_Up>&, _Args&& ...>, _Tp&> std::optional<_Tp>::emplace(std::initializer_list<_Up>, _Args&& ...) [with _Up = _Up; _Args = {_Args ...}; _Tp = grpc_core::testing::{anonymous}::XdsClientTest::FakeXdsBootstrap::FakeNode]’
  855 |  emplace(initializer_list<_Up> __il, _Args&&... __args)
      |  ^~~~~~~
/usr/include/c++/9/optional:855:2: note:   template argument deduction/substitution failed:
/root/projects/grpc/grpc/test/core/xds/xds_client_test.cc:164:47: note:   candidate expects at least 1 argument, 0 provided
  164 |         if (!node_.has_value()) node_.emplace();
      |                                               ^
```

--- a/test/core/xds/xds_client_test.cc
+++ b/test/core/xds/xds_client_test.cc
@@ -82,6 +82,7 @@ class XdsClientTest : public ::testing::Test {
    public:
     class FakeNode : public Node {
      public:
+      FakeNode() = default;
       const std::string& id() const override { return id_; }
       const std::string& cluster() const override { return cluster_; }
       const std::string& locality_region() const override {

